src/Numeric/Units/Dimensional/Dynamic.hs

```diff
@@ -101,19 +101,23 @@ instance Num a => Monoid (AnyQuantity a) where
 -- not be a 'Quantity' of any 'Dimension' whatsoever, but in exchange it gains instances
 -- for the common numeric classes. It's therefore useful for manipulating, and not merely storing,
 -- quantities of unknown dimension.
-newtype DynQuantity a = DynQuantity (Maybe (AnyQuantity a))
-  deriving (Eq, Data, Generic, Generic1, Typeable, Show)
+data DynQuantity a = GoodQuantity (AnyQuantity a) | BadQuantity
+  deriving (Data, Generic, Generic1, Typeable, Show)
+
+instance Eq a => Eq (DynQuantity a) where
```

Comment:
I'm confused as to why this isn't still derivable even for the explicit definition?